pkg/server/server.go

```diff
@@ -0,0 +1,90 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/replicatedhq/ekco/pkg/ekcoops"
+	"github.com/replicatedhq/pvmigrate/pkg/migrate"
+)
+
+const (
+	MIGRATION_STATUS_RUNNING   = "running"
+	MIGRATION_STATUS_FAILED    = "failed"
+	MIGRATION_STATUS_COMPLETED = "completed"
+)
+
+var migrateStorageMut = sync.Mutex{}
+var migrationStatus = ""
+var migrationLogs = ""
+
+func Serve(config ekcoops.Config, client kubernetes.Interface) {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	http.HandleFunc("/storagemigration/status", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(migrationStatus))
+		if err != nil {
+			log.Printf("failed to write status: %v", err)
+		}
+	})
+
+	http.HandleFunc("/storagemigration/logs", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(migrationLogs))
+		if err != nil {
+			log.Printf("failed to write logs: %v", err)
+		}
+	})
+
+	http.HandleFunc("/storagemigration/approve", func(w http.ResponseWriter, r *http.Request) {
+		go migrateStorage(config, client)
```

Comment:
If we want pass parameters that it is:

```
		from := r.URL.Query().Get("from")
		if from == "" {
			from = "scaling"
		}

		_, err := w.Write([]byte(fmt.Sprintf("Starting migration from (%s) to (%s) storage class", from)))
```

Reply:
yeah
I think I'm 100% going to leave that for a "later" thing